modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/reconnection/ReconnectionTestCase.java

```diff
@@ -106,13 +108,13 @@ public void getInlineRetryPolicyTemplate() throws Exception {
   public void reconnectAfterConnectionExceptionOnFirstPage() throws Exception {
     resetCounters();
     Iterator<ReconnectableConnection> iterator = getCursor("pagedOperation", 1, CONNECTIVITY);
-    ReconnectableConnection firstPage = iterator.next();
-    assertThat("Connection was not disconnected.", firstPage.getDisconnectCalls(), is(1));
+    iterator.next();
+    assertThat("Connection was not disconnected.", disconnectCalls, is(1));
     assertThat("Paging provider was not closed.", closePagingProviderCalls, is(1));
   }
 
-  @Test
-  public void doNotReconnectAfterOtherExceptionOnFirstPage() {
+  @Test(expected = IllegalArgumentException.class)
```

Comment:
use the ExpectedException rule and assert on the cause being THE SAME instance you are expecting and the actual exception being of the correct type

Reply:
I tried the approach with ExpectedException, but it does not allow me to assert other things after the exception is thrown (like the paging provider closing or the disconnection calls).